tools/crm_mon_print.c

```diff
@@ -54,6 +54,48 @@ static int print_node_attributes(pcmk__output_t *out, pe_working_set_t *data_set
 static int print_failed_actions(pcmk__output_t *out, pe_working_set_t *data_set,
                                 gboolean print_spacer);
 
+static gboolean
+str_in_unames(gchar **unames, const gchar *s) {
+    return unames && g_strv_contains((const gchar * const*) unames, s);
+}
+
+static gchar **
+build_uname_list(pe_working_set_t *data_set, const char *s) {
+    gchar **unames = NULL;
+
+    if (s == NULL) {
+        /* Nothing was given so return a list of all node names. */
+        int i = 0;
+        unames = calloc(g_list_length(data_set->nodes)+1, sizeof(gchar *));
+
+        for (GListPtr iter = data_set->nodes; iter != NULL; iter = iter->next) {
+            pe_node_t *node = (pe_node_t *) iter->data;
+            unames[i] = (gchar *) strdup(node->details->uname);
```

Comment:
Since you're using g_strfreev() to free it, you need to use g_strdup((gchar *) ...) to allocate

Reply:
Argh, I keep doing this.  I'd love to figure out something I could run to check for these automatically.

Reply:
Yep, that's why at one point we got rid of all our usage of gchar. :) That was easier than keeping the two straight. Unfortunately there's no easy equivalent for the strv stuff. Probably the closest alternative would be to use a GList of char* instead of strv's.

Reply:
If you think it's worth it, I could convert pe__unames_with_tag to return a GListPtr and I could also probably get rid of the GPtrArray intermediate step.

Reply:
Yeah I think this it worth it, if for no other reason than it gets rid of those ridiculous type casts in lib/fencing/st_output.c.